integration-tests/src/test/java/ch/bifrost/integrationtest/session/DefaultSessionTest.java

```diff
@@ -0,0 +1,72 @@
+package ch.bifrost.integrationtest.session;
+
+import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.Level;
+import org.apache.log4j.PatternLayout;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ch.bifrost.client.impl.session.ClientSessionEndpoint;
+import ch.bifrost.client.impl.session.DefaultSessionLayerClientAdapter;
+import ch.bifrost.client.impl.session.DefaultSessionLayerClientAdapter.DefaultSessionLayerClientAdapterFactory;
+import ch.bifrost.core.api.session.Message;
+import ch.bifrost.server.impl.session.DefaultSessionLayerServerAdapter.DefaultSessionLayerServerAdapterFactory;
+import ch.bifrost.server.impl.session.EchoServer.EchoServerFactory;
+import ch.bifrost.server.impl.session.MultiplexingSessionAdapter;
+
+public class DefaultSessionTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultSessionTest.class);
+	
+	private static final int SERVER_PORT = 12345;
+	private MultiplexingSessionAdapter server;
+	private ClientSessionEndpoint<DefaultSessionLayerClientAdapter> client;
+
+	@BeforeClass
+	public static void setupLogger() {
+		ConsoleAppender console = new ConsoleAppender();
+		String pattern = "%d [%-5p|%c] %m%n";
+		console.setLayout(new PatternLayout(pattern));
+		console.setThreshold(Level.ALL);
+		console.activateOptions();
+		org.apache.log4j.Logger.getRootLogger().addAppender(console);
+	}
+	
+	@Before
+	public void setupClientAndServer() throws Exception {
+		server = new MultiplexingSessionAdapter(SERVER_PORT, new DefaultSessionLayerServerAdapterFactory(), new EchoServerFactory());
```

Comment:
Ja, das wäre sauberer und konzeptionell einfacher, ich hab es aber nicht hinbekommen, insbesondere auf dem Server. Dort wird der Stack in umgekehrter Reihenfolge aufgebaut (zumindest ab dem ersten Multiplexer): Wenn eine neue Session kreiert wird, muss ein kompletter Stack von unten nach oben aufgebaut werden.

Meine Idee hier war eine Art "Inversion of Control" (stimmt nicht ganz): Statt dem oberen Layer einen unteren Layer mitzugeben, gebe ich dem unteren Layer eine Factory mit, über die es die nächst obere Schicht erstellen kann sobald es nötig ist. In diesem Schritt gibt sich die untere Schicht dann der über die Factory neu erstellen oberen Schicht selber mit. Schlussendlich hätte man dadurch wieder das Gerüst, wie du es beschreibst, nur irgendwie andersrum. Keine Ahnung ob diese Erklärung jetzt hilfreich war :D...